tests/func/test_update.py

```diff
@@ -19,8 +19,13 @@ def test_update_import(tmp_dir, dvc, erepo_dir):
     # cli call, so we need to clean the caches to see the changes.
     clean_repos()
 
-    assert dvc.status([stage.path]) == {}
+    status, = dvc.status([stage.path])["version.dvc"]
+    changed_dep, = list(status["changed deps"].items())
+    assert changed_dep[0].startswith("version ")
+    assert changed_dep[1] == "update available"
+
     dvc.update(stage.path)
+
```

Comment:
This one only tests cached files, but we actually have a bug described in https://github.com/iterative/dvc/issues/3087#issuecomment-574267505 that will make the test break for git files 🙁I've submitted https://github.com/iterative/dvc/pull/3151 , that will unblock you.

Reply:
@fabiosantoscode Could you please add a cross test though? The one where file gets updated from being git tracked to dvc tracked.

Reply:
So that's why rev_lock was showing up inconsistently on my test files :)

Reply:
@fabiosantoscode Merged and unblocked. Thank you for your patience 🙂

Reply:
Super cool @efiop :) 

So this cross-test.

I think you mean I need to add a test that imports from a DVC project a file that's not under DVC (but is under git), then track the file under DVC and do an update?

Or is it an import before `dvc init` and an update after?

Maybe both? :)

Reply:
@fabiosantoscode Good question. What I meant is:
1) import git-tracked `file` from erepo
2) make `file` dvc-tracked in erepo, create a commit
3) run `dvc update`

So your first description, looks like.

> Or is it an import before dvc init and an update after?

But that is indistinguishable from the previous description from dvc's point of view 🙂Unless I'm missing something. But the scenario is more believable than mine, for sure.

Reply:
@efiop after I got the first one passing, I tried the second one just to see if it passed.

It failed, and revealed that this PR doesn't allow `status` for imported files from non-git repositories.

So will need to do more stuff :) but it's getting late. Will do some more of this tomorrow!